summarization/summarizer_main.py

```diff
@@ -202,7 +202,9 @@ def _assemble_and_instantiate_stamp_page(
         stamp_position = -1
 
         if text:
-            sentence_index = text.index
+            # casting is necessary here since
```

Comment:
Lets have 2 attributes paragraph_index, and sentence_index

Reply:
We can have something like this
```
get_weight():
        return paragraph_index + sentence_index * sentence_unit_weight
```

Reply:
That sounds great! amending accordingly!